tests/test_filtsmooth/test_gaussfiltsmooth/test_kalman.py

```diff
@@ -50,3 +50,127 @@ def test_info_dicts(setup):
+    smooms = posterior.states.mean
+
+    filtms_rmse = np.mean(np.abs(filtms[:, :2] - truth[:, :2]))
+    smooms_rmse = np.mean(np.abs(smooms[:, :2] - truth[:, :2]))
+    obs_rmse = np.mean(np.abs(regression_problem.observations - truth[:, :2]))
+
+    assert smooms_rmse < filtms_rmse < obs_rmse
+
+
+def test_kalman_smoother_high_order_ibm_with_process_noise_damping_crazy_small_steps():
+    """If we add a small damping factor to the computation of the Cholesky factor of the
+    process noise covariance, we can do order 15 on the car tracking model at least."""
+    regression_problem, statespace_components = filtsmooth_zoo.car_tracking(
+        model_ordint=14,
+        timespan=(0.0, 1e-10),
+        step=1e-12,
+        forward_implementation="sqrt",
+        backward_implementation="sqrt",
+        _process_noise_damping=1e-14,
+    )
+    truth = regression_problem.solution
+
+    kalman = filtsmooth.Kalman(
+        statespace_components["dynamics_model"],
+        statespace_components["measurement_model"],
+        statespace_components["initrv"],
+    )
+
+    posterior, _ = kalman.filtsmooth(regression_problem)
+
+    filtms = posterior.filtering_posterior.states.mean
+    smooms = posterior.states.mean
+
+    filtms_rmse = np.mean(np.abs(filtms[:, :2] - truth[:, :2]))
+    smooms_rmse = np.mean(np.abs(smooms[:, :2] - truth[:, :2]))
+    obs_rmse = np.mean(np.abs(regression_problem.observations - truth[:, :2]))
+
+    assert smooms_rmse < filtms_rmse < obs_rmse
+
+
+def test_kalman_smoother_high_order_ibm_with_process_noise_damping():
+    """If we add a small damping factor to the computation of the Cholesky factor of the
+    process noise covariance, we can do order 15 on the car tracking model at least."""
+    regression_problem, statespace_components = filtsmooth_zoo.car_tracking(
+        model_ordint=15,
+        timespan=(0.0, 0.1),
+        step=1e-3,
+        forward_implementation="sqrt",
+        backward_implementation="sqrt",
+        _process_noise_damping=1e-15,
+    )
+    truth = regression_problem.solution
+
+    kalman = filtsmooth.Kalman(
+        statespace_components["dynamics_model"],
+        statespace_components["measurement_model"],
+        statespace_components["initrv"],
+    )
+
+    posterior, _ = kalman.filtsmooth(regression_problem)
+
+    filtms = posterior.filtering_posterior.states.mean
+    smooms = posterior.states.mean
+
+    filtms_rmse = np.mean(np.abs(filtms[:, :2] - truth[:, :2]))
+    smooms_rmse = np.mean(np.abs(smooms[:, :2] - truth[:, :2]))
+    obs_rmse = np.mean(np.abs(regression_problem.observations - truth[:, :2]))
+
+    assert smooms_rmse < filtms_rmse < obs_rmse
+
+
+def test_kalman_filter_high_order_ibm_with_process_noise_damping():
+    """With damping, the filter achieves really high orders (we test 25, but 50 was
+    possible too, for some reason)."""
+    regression_problem, statespace_components = filtsmooth_zoo.car_tracking(
+        model_ordint=25,
```

Comment:
What is happening.

Reply:
Just for the jokes, the test passes for order=75 as well (no typo!), but I thought 25 was sufficiently extreme :D

Reply:
All just by removing the matrix inversion, right ?

Reply:
and by using this mini damping factor that makes the process noise always positive definite.